sdk/nodejs/runtime/rpc.ts

```diff
@@ -334,10 +342,14 @@ export async function serializeProperty(ctx: string, prop: Input<any>, dependent
         // and tracked in a reasonable manner, while not causing us to compute or embed information
         // about it that is not needed, and which can lead to deadlocks.
         if (excessiveDebugOutput) {
-            log.debug(`Serialize property [${ctx}]: component resource urnid`);
+            log.debug(`Serialize property [${ctx}]: component resource urn`);
         }
 
-        return serializeProperty(`${ctx}.urn`, prop.urn, dependentResources);
+        const urn = await serializeProperty(`${ctx}.urn`, prop.urn, dependentResources);
+        return {
+            [specialSigKey]: specialResourceSig,
+            urn: urn,
+        };
```

Comment:
since Custom and Component resources are the same now, we can unify these.

Reply:
Yes - except that we'll likely instead need to keep these conditional to avoid the breaking change.